cmd/applyconfig/applyconfig.go

```diff
@@ -239,6 +242,55 @@ func applyConfig(rootDir, cfgType string, process processFn) error {
 	return nil
 }
 
+type censoringFormatter struct {
+	secrets  sets.String
+	delegate logrus.Formatter
+}
+
+func (f censoringFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	raw, err := f.delegate.Format(entry)
+	if err != nil {
+		return raw, err
+	}
+
+	return f.Censor(raw), nil
+}
+
+const censored = "CENSORED"
+
+var censoredBytes = []byte(censored)
+
+func (f *censoringFormatter) Censor(content []byte) []byte {
+	if f.secrets != nil {
+		for _, secret := range f.secrets.List() {
+			content = bytes.ReplaceAll(content, []byte(secret), censoredBytes)
+		}
+	}
+	return content
+}
+
+func getCensoringFormatter(f logrus.Formatter) logrus.Formatter {
+	secrets := sets.NewString()
+	//those strings do not look like valid secrets
+	noSecrets := sets.NewString("", ":", "1")
+	for _, e := range os.Environ() {
+		pair := strings.SplitN(e, "=", 2)
+		if noSecrets.Has(pair[1]) {
+			logrus.WithField("key", pair[0]).WithField("value", pair[1]).Warn("ignoring censoring value to make log useful ...")
```

Comment:
This is from the test on my PC:

```
$  ./applyconfig --config-dir /tmp/test_ac
WARN[0000] ignoring censoring value to make log useful ...  key=MANPATH value=":"
WARN[0000] ignoring censoring value to make log useful ...  key=LOADEDMODULES value=
WARN[0000] ignoring censoring value to make log useful ...  key=SHLVL value=1

```

It indicates that any secret in `sets.NewString("", ":", "1")` will be leaked.
Those are not acceptable as secret anyway.

Reply:
Why do we need an approach that is any different from the upstream `bytes.Replace`?

Reply:
I probably wont need this when fulfil https://github.com/openshift/ci-tools/pull/224/files#r336988964